packages/grafbase-sdk/src/type.ts

```diff
@@ -0,0 +1,41 @@
+import { Field } from "./field"
+import { GListDef } from "./field/list"
+import { GScalarDef } from "./field/typedefs"
+import { Interface } from "./interface"
+
+export class Type {
+  name: string
+  fields: Field[]
+  interface?: Interface
+
+  constructor(name: string) {
+    this.name = name
+    this.fields = []
+  }
+
+  public field(name: string, definition: GScalarDef | GListDef): Type {
+    this.fields.push(new Field(name, definition))
+
+    return this
+  }
+
+  public implements(i: Interface): Type {
+    this.interface = i
+
+    return this
+  }
+
+  public toString(): string {
+    const impl = this.interface ? ` implements ${this.interface?.name}` : ""
+    const header = `type ${this.name}${impl} {`
```

Comment:
types can implement multiple interfaces, so you might want to make `this.interface` a list